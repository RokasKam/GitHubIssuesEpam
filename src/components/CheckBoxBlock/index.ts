import { CheckBoxBlock } from "./CheckBoxBlock";
export { CheckBoxBlock };