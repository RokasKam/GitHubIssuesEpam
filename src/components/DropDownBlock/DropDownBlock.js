import React, { useState } from 'react'
import { View, Text } from 'react-native';
import DropDownPicker from 'react-native-dropdown-picker';
import { styling } from './DropDownBlock.style';

export const DropDownBlock = ({ value, setValue }) => {

    const [open, setOpen] = useState(false);
    const styles = styling(open);
    const items = [
        { label: 'Created', value: 'created' },
        { label: 'Updated', value: 'updated' },
        { label: 'Popularity', value: 'popularity' },
        { label: 'Long-running', value: 'long-runing' }
    ];

    return (
        <View style={styles.blockView}>
            <Text style={styles.text}>How to sort</Text>
            <DropDownPicker
                style={styles.dropDown}
                textStyle={styles.dropDownText}
                containerStyle={styles.dropDownContainer}
                dropDownContainerStyle={styles.dropDownList}
                open={open}
                value={value}
                items={items}
                setOpen={setOpen}
                setValue={setValue}
            />
        </View>

    )
};

