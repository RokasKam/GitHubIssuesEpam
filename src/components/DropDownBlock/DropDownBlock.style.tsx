import { StyleSheet } from 'react-native';

export const styling = (open: boolean) =>
    StyleSheet.create({
        blockView: {
            alignItems: "center",
        },
        dropDown: {
            backgroundColor: "#00bf33",
            width: "80%",
        },
        dropDownContainer: {
            height: open ? 210 : undefined,
            marginBottom: 20,
        },
        dropDownList: {
            backgroundColor: "#00bf33",
            width: "80%",
        },
        dropDownText: {
            fontSize: 15,
            color: "white"
        },
        text: {
            color: "#323C4C",
            marginBottom: 10,
            fontSize: 15
        }
    });