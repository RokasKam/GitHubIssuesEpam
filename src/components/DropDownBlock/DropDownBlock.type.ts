export type Props = {
    value: string;
    setValue: React.Dispatch<React.SetStateAction<string>>;
};