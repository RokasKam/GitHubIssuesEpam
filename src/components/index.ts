export { CheckBoxBlock } from "./CheckBoxBlock";
export { DropDownBlock } from "./DropDownBlock";
export { IssuesList } from "./IssuesList";
export { ListCard } from "./ListCard";
export { SearchButton } from "./SearchButton";
export { SearchInput } from "./SearchInput";

