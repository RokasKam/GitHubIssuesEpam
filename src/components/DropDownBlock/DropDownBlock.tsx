import React, { useState } from 'react'
import { View, Text } from 'react-native';
import DropDownPicker from 'react-native-dropdown-picker';
import { styling } from './DropDownBlock.style';
import { Props } from './DropDownBlock.type';

export const DropDownBlock: React.FC<Props> = ({ value, setValue }) => {

    const [open, setOpen] = useState(false);
    const styles = styling(open);
    const items = [
        { label: 'Newest', value: 'created' },
        { label: 'Recently updated', value: 'updated' },
        { label: 'Most commented', value: 'comments' },
        { label: 'Most reactions', value: 'reactions' }
    ];

    return (
        <View style={styles.blockView}>
            <Text style={styles.text}>How to sort</Text>
            <DropDownPicker
                style={styles.dropDown}
                textStyle={styles.dropDownText}
                containerStyle={styles.dropDownContainer}
                dropDownContainerStyle={styles.dropDownList}
                open={open}
                value={value}
                items={items}
                setOpen={setOpen}
                setValue={setValue}
            />
        </View>

    )
};

