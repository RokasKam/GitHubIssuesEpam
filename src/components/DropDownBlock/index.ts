import { DropDownBlock } from "./DropDownBlock";
export { DropDownBlock };