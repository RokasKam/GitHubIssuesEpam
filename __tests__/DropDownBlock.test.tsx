import React, { SetStateAction } from 'react';
import renderer, { act } from 'react-test-renderer';
import { DropDownBlock } from '../src/components/DropDownBlock'

test('renders correctly', async () => {
  const tree = renderer.create(<DropDownBlock value={''} setValue={function (value: SetStateAction<string>): void {
    throw new Error('Function not implemented.');
  }} />).toJSON();
  await act(async () => {
    expect(tree).toMatchSnapshot();
  });
});