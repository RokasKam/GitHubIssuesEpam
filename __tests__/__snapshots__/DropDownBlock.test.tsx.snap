// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323C4C",
        "fontSize": 15,
        "marginBottom": 10,
      }
    }
  >
    How to sort
  </Text>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "zIndex": 5000,
        },
        Object {
          "height": undefined,
          "marginBottom": 20,
        },
      ]
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onLayout={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00bf33",
          "borderColor": "#000",
          "borderRadius": 8,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "minHeight": 50,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 3,
          "width": "80%",
          "zIndex": 5000,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
            },
            Object {
              "color": "white",
              "fontSize": 15,
            },
            false,
            Object {},
          ]
        }
      >
        Select an item
      </Text>
      <View
        style={
          Array [
            Object {
              "marginLeft": 10,
            },
            Object {},
          ]
        }
      >
        <Image
          source="arrow-down.png"
          style={
            Array [
              Object {
                "height": 20,
                "width": 20,
              },
              Object {},
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
